wicket-user-guide/src/main/asciidoc/security/security_5.adoc

```diff
@@ -2,7 +2,40 @@
 
 _CryptoMapper_ helps preventing CSRF attacks by making the urls impossible to be guessed by an attacker but still there is some theoretical chance this to happen.
 
-To further help against this kind of vulnerability Wicket provides _CsrfPreventionRequestCycleListener_ - a _IRequestCycleListener_ that forbids requests made from a different origin. By default only actions are forbidden, i.e. a request coming from different origin cannot execute _Link.onClick()_ or submit forms (_Form.onSubmit()_). Any request to render pages are still allowed so Wicket pages could be easily embedded in other applications.
+To further help against this kind of vulnerability Wicket provides _ResourceIsolationRequestCycleListener_ - a _IRequestCycleListener_ that uses __IResourceIsolationPolicy__ objects to decide whether to allow or reject a request. By default only actions are checked, i.e. a request coming from different origin cannot execute _Link.onClick()_ or submit forms (_Form.onSubmit()_). Any request to render pages are still allowed so Wicket pages could be easily embedded in other applications.
+
+MyApplication.java
+[source,java]
+----
+  @Override
+ protected void init() {
+  super.init();
+  getRequestCycleListeners().add(new ResourseIsolationRequestCycleListener());
+  // ...
+ }
+----
+
+_ResourceIsolationRequestCycleListener_ is highly configurable. It allows to add exempted paths that will not be checked with the __addExemptedPath__ method. It can be configured with multiple _ResourceIsolationPolicy_ objects to be checked in order.
+
+An __IResourceIsolationPolicy__ returns a __ResourceIsolationOutcome__ after processing a request, which can be one of 3 values (__ALLOWED__, __DISALLOWED__, __UNKNOWN__). The __ResourceIsolationRequestCycleListener__ checks the __IResourceIsolationPolicy__ objects in order and uses the first outcome that is not __UNKNOWN__ to trigger the appropriate action. If all return __UNKNOWN__ __unknownOutcomeAction__ is applied. The actions can be configured through the listener.
```

Comment:
what if an outcome returns ALLOWED? we keep iterating over the remaining policies, right? If so, this should read: 'the first outcome that is not __UNKNOWN__ or __ALLOWED__', I believe

Reply:
I think if the outcome is `ALLOWED` we [stop iterating over the remaining policies and return](https://github.com/apache/wicket/blob/0f42d3314f1200a3b88f76e435910ef389af6b58/wicket-core/src/main/java/org/apache/wicket/protocol/http/ResourceIsolationRequestCycleListener.java#L336-L339)

Reply:
right! not our brightest moment 🤦‍♂️